Rajitha/ItemSales.js

```diff
@@ -0,0 +1,146 @@
+addCustomerSales('Jon','Bandaid',5);
+
+console.log(customerSales);
+
+/***********************************************************
+ *How many items were scanned before Jon's Hat got scanned?
+ ***********************************************************/
+const scanItemsBeforeCustomer = (customer,order) => {
+    for(let i=0; i<customerSales.length; i++){
+        if (customerSales[i].Customer == customer && customerSales[i].Order === order) {
+            return i++;
+        }
+    }
+}
+console.log("Number of items scanned before Jon are " +scanItemsBeforeCustomer('Jon','Hat'));
+
+/*********************************************
+ *      How many people bought socks
+ *********************************************/
+const socksCustomerCount = (order) => {
+    let customerCount = 0;
+    for(let i=0; i<customerSales.length; i++){
+        if (customerSales[i].Order == order){
+            customerCount++;
+        }
+    }
+    return customerCount;
+}
+console.log("Number of people who bought socks are " +socksCustomerCount('Socks'));
+
+/*********************************************
+ *      Total Quantity Sold
+ *********************************************/
+const totalSoldQuantity = () => {
+    let totalQuantity = 0;
+    for (let i=0; i<customerSales.length; i++){
+        totalQuantity += customerSales[i].Quantity;
+    }
+    return totalQuantity;
+}
+console.log("Total Quantity Sold " +totalSoldQuantity());
+
+/*********************************************
+ *      Did Sally buy a dress
+ *********************************************/
+const checkIfDressBought = (customer,order) => {
+    for (let i=0; i<customerSales.length; i++){
+        if((customerSales[i].Customer === customer) && (customerSales[i].Order === order))
+            return true;
+    }
+}
+console.log("Did Sally buy a Dress: " );
+console.log(checkIfDressBought('Sally','Dress')?"Yes":"No");
+
+console.log("Did Dan buy a Dress: " );
+console.log(checkIfDressBought('Dan','Dress')?"Yes":"No");
+
+/*********************************************
+ * What is the total amount after tax?
+ *********************************************/
+/** Calculate itemized price and add the details to new array **/
+let itemsPrice = [];
+const itemizedPriceSummary = () => {
+    let totalSingleOrderCount = 0;
+    for(let j=0;j<items.length;j++)
+    {
+        for (let i=0;i<customerSales.length;i++) {
+            if (items[j].Item === customerSales[i].Order) {
+                totalSingleOrderCount += customerSales[i].Quantity;
+            }
+        }
+        let totalItemPriceBeforeTax = totalSingleOrderCount * items[j].Price;
+        let totalItemPriceAfterTax = (totalItemPriceBeforeTax + (totalItemPriceBeforeTax * (tax[0].Tax/100)));
+        itemsPrice.push({'ItemQuantity': totalSingleOrderCount, 'PriceBeforeTax': totalItemPriceBeforeTax, 'PriceAfterTax': totalItemPriceAfterTax});
+        totalSingleOrderCount = 0;
+    }
+    return items;
+}
```

Comment:
Can you try by breaking down this into multiple functions instead of nested for loops?